Return an error when trying to execute a task on a non running service

antho1404 opened the pull request:
When calling the `executeTask` API, if the service was not running the system was adding the task in the queue but this task would have no chances to be executed. 
Now the API will return an error if the service is not running

fixes #307

krhubert commented on api/core/error.go:
```diff
@@ -0,0 +1,10 @@
+package core
+
+// NotRunningServiceError is an error when a service is not running
```
Small change, missing dot at the end.

ilgooz commented on api/core/execute.go:
```diff
@@ -6,25 +6,50 @@ import (
+	if err != nil {
+		return "", err
+	}
+	err = exc.Execute()
```
please add this line directly to return

ilgooz commented:
lgtm

ilgooz commented on api/core/execute.go:
```diff
@@ -6,25 +6,50 @@ import (
 	}
-	execution, err := execution.Create(&service, request.TaskKey, inputs)
-	if err != nil {
+	if err := checkService(&srv); err != nil {
```
checkService() before getData()

ilgooz commented:
on CI:
```
--- FAIL: TestExecuteWithInvalidJSON (0.02s)
	assert.go:20: /go/src/github.com/mesg-foundation/core/api/core/execute_test.go:57
	assert.go:29: 💩  "Service v1_f7a73ba457db8de2f187635fe8bbdb47 is not running" != "unexpected end of JSON input"
```

antho1404 commented:
Tests are ok now, I moved the `checkService` and renamed it into `checkServiceStatus` to be more precise. This check can be done either before or after the `getData`, I chose to put it after just for optimizations, `checkServiceStatus` make an api call to docker so if there is an error with the data at least we save one call

ilgooz commented:
@antho1404 makes sense, we should always validate data before an external call
